test_store_cleanup_disk_s3 is flaky

https://s3.amazonaws.com/clickhouse-test-reports/0/c3482a104f73f237450f7d7781871859472687d9/integration_tests__asan__[6/6].html

https://play.clickhouse.com/play?user=play#U0VMRUNUIHB1bGxfcmVxdWVzdF9udW1iZXIsIGNoZWNrX3N0YXJ0X3RpbWUsIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgdGVzdF9zdGF0dXMsIGNoZWNrX3N0YXR1cywgcmVwb3J0X3VybApGUk9NIGNoZWNrcwpXSEVSRSAxCiAgICAtLUFORCBwdWxsX3JlcXVlc3RfbnVtYmVyID0gMAogICAgQU5EIHRlc3Rfc3RhdHVzICE9ICdTS0lQUEVEJwogICAgQU5EIHRlc3Rfc3RhdHVzICE9ICdPSycKICAgIEFORCBjaGVja19zdGF0dXMgIT0gJ3N1Y2Nlc3MnCiAgICBBTkQgdGVzdF9uYW1lIGxpa2UgJyV0ZXN0X3N0b3JlX2NsZWFudXBfZGlza19zMyUnCk9SREVSIEJZIGNoZWNrX3N0YXJ0X3RpbWUgZGVzYywgY2hlY2tfbmFtZSwgdGVzdF9uYW1l

Hi. 
This is helping a bit.
https://github.com/ClickHouse/ClickHouse/pull/49157 
At least no more diff in s3.
But now the different error unveal.

https://s3.amazonaws.com/clickhouse-test-reports/49157/ff648b7b36aa2d45f52c0e2cd28c90810a502c69/integration_tests__release__[2_4].html

```
E           helpers.client.QueryRuntimeException: Client failed! Return code: 117, stderr: Received exception from server (version 23.5.1):
E           Code: 117. DB::Exception: Received from 172.16.1.6:9000. DB::Exception: Data directory for table already containing data parts - probably it was unclean DROP table or manual intervention. You must either clear directory by hand or use ATTACH TABLE instead of CREATE TABLE if you need to use that parts.. Stack trace:
E           
E           0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0xe38e8b5 in /usr/bin/clickhouse
E           1. ? @ 0x14b2eee4 in /usr/bin/clickhouse
E           2. DB::StorageMergeTree::StorageMergeTree(DB::StorageID const&, String const&, DB::StorageInMemoryMetadata const&, bool, std::shared_ptr<DB::Context>, String const&, DB::MergeTreeData::MergingParams const&, std::unique_ptr<DB::MergeTreeSettings, std::default_delete<DB::MergeTreeSettings>>, bool) @ 0x14b2ed65 in /usr/bin/clickhouse
E           3. ? @ 0x14b2b568 in /usr/bin/clickhouse
E           4. DB::StorageFactory::get(DB::ASTCreateQuery const&, String const&, std::shared_ptr<DB::Context>, std::shared_ptr<DB::Context>, DB::ColumnsDescription const&, DB::ConstraintsDescription const&, bool) const @ 0x1427859b in /usr/bin/clickhouse
E           5. DB::InterpreterCreateQuery::doCreateTable(DB::ASTCreateQuery&, DB::InterpreterCreateQuery::TableProperties const&, std::unique_ptr<DB::DDLGuard, std::default_delete<DB::DDLGuard>>&) @ 0x139593f1 in /usr/bin/clickhouse
E           6. DB::InterpreterCreateQuery::createTable(DB::ASTCreateQuery&) @ 0x13952162 in /usr/bin/clickhouse
E           7. DB::InterpreterCreateQuery::execute() @ 0x1395e8d4 in /usr/bin/clickhouse
E           8. ? @ 0x13ee1007 in /usr/bin/clickhouse
E           9. DB::executeQuery(String const&, std::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum) @ 0x13ede58d in /usr/bin/clickhouse
E           10. DB::TCPHandler::runImpl() @ 0x14cb93f1 in /usr/bin/clickhouse
E           11. DB::TCPHandler::run() @ 0x14cceef9 in /usr/bin/clickhouse
E           12. Poco::Net::TCPServerConnection::start() @ 0x17c42314 in /usr/bin/clickhouse
E           13. Poco::Net::TCPServerDispatcher::run() @ 0x17c4353b in /usr/bin/clickhouse
E           14. Poco::PooledThread::run() @ 0x17dc08e7 in /usr/bin/clickhouse
E           15. Poco::ThreadImpl::runnableEntry(void*) @ 0x17dbe31d in /usr/bin/clickhouse
E           16. ? @ 0x7fec1e612609 in ?
E           17. clone @ 0x7fec1e537133 in ?
E           . (INCORRECT_DATA)
E           (query: CREATE TABLE s3_test UUID '00000000-1000-4000-8000-000000000001' (n UInt64) Engine=MergeTree() ORDER BY n SETTINGS storage_policy='s3';)
```

--- a/tests/integration/test_merge_tree_s3/test.py
+++ b/tests/integration/test_merge_tree_s3/test.py
@@ -740,31 +740,6 @@ def test_cache_with_full_disk_space(cluster, node_name):
 
 
 @pytest.mark.parametrize("node_name", ["node"])
-def test_store_cleanup_disk_s3(cluster, node_name):
-    node = cluster.instances[node_name]
-    node.query("DROP TABLE IF EXISTS s3_test SYNC")
-    node.query(
-        "CREATE TABLE s3_test UUID '00000000-1000-4000-8000-000000000001' (n UInt64) Engine=MergeTree() ORDER BY n SETTINGS storage_policy='s3';"
-    )
-    node.query("INSERT INTO s3_test SELECT 1")
-
-    node.stop_clickhouse(kill=True)
-    path_to_data = "/var/lib/clickhouse/"
-    node.exec_in_container(["rm", f"{path_to_data}/metadata/default/s3_test.sql"])
-    node.start_clickhouse()
-
-    node.wait_for_log_line(
-        "Removing unused directory", timeout=90, look_behind_lines=1000
-    )
-    node.wait_for_log_line("directories from store")
-    node.query(
-        "CREATE TABLE s3_test UUID '00000000-1000-4000-8000-000000000001' (n UInt64) Engine=MergeTree() ORDER BY n SETTINGS storage_policy='s3';"
-    )
-    node.query("INSERT INTO s3_test SELECT 1")
-    check_no_objects_after_drop(cluster)
-
-
-@pytest.mark.parametrize("node_name", ["node"])
 def test_cache_setting_compatibility(cluster, node_name):
     node = cluster.instances[node_name]
 

